planning/behavior_velocity_planner/src/scene_module/occlusion_spot/occlusion_spot_utils.cpp

```diff
@@ -157,9 +157,12 @@ void clipPathByLength(
   const PathWithLaneId & path, PathWithLaneId & clipped, const double max_length)
 {
   double length_sum = 0;
-  for (int i = 0; i < static_cast<int>(path.points.size()) - 1; i++) {
-    length_sum += tier4_autoware_utils::calcDistance2d(path.points.at(i), path.points.at(i + 1));
+  auto p0 = path.points.at(0);
+  for (int i = 0; i < static_cast<int>(path.points.size()); i++) {
+    const auto & p1 = path.points.at(i);
+    length_sum += tier4_autoware_utils::calcDistance2d(p0, p1);
     if (length_sum > max_length) return;
+    p0 = p1;
     clipped.points.emplace_back(path.points.at(i));
   }
```

Comment:
I prefer to implement like this
``` double length_sum = 0;
  for (int i = 0; i < static_cast<int>(path.points.size()) - 1; i++) {
    length_sum += tier4_autoware_utils::calcDistance2d(path.points.at(i), path.points.at(i + 1));
    if (length_sum > max_length) return;
    clipped.points.emplace_back(path.points.at(i));
  }
  clipped.points.emplace_back(path.points.back());
```

Reply:
If you don't agree, please ignore

Reply:
@takayuki5168 
this is a place where path point is clipped so to do this it's necessary to have counter outside loop for the last point.
so this will be like 

```
int last_point = 0;
  for (int i = 0; i < static_cast<int>(path.points.size()) - 1; i++) {
    length_sum += tier4_autoware_utils::calcDistance2d(path.points.at(i), path.points.at(i + 1));
    if (length_sum > max_length) return;
    clipped.points.emplace_back(path.points.at(i));
    last_point = i;
  }
  clipped.points.emplace_back(path.points.at(last_point));
```

Reply:
![image](https://user-images.githubusercontent.com/65527974/157567079-1e733910-143e-4040-9ffa-8c686f85a6cf.png)

Reply:
@taikitanaka3 Ahh soryy, I wrote wrong. How about this one?
``` 
clipped.points.emplace_back(path.points.front());
for (int i = 1; i < static_cast<int>(path.points.size()); ++i) {
    const double length_sum += tier4_autoware_utils::calcDistance2d(path.points.at(i), path.points.at(i - 1));
    if (length_sum > max_length) return;
    clipped.points.emplace_back(path.points.at(i));
}
```

Reply:
very nice!

Reply:
![image](https://user-images.githubusercontent.com/65527974/157570394-36b8731c-315a-45c6-a637-b0d9da1e685e.png)